src/errorpage.cc

```diff
@@ -142,44 +142,49 @@ static void ValidateStaticError(const int page_id, const SBuf &inputLocation);
 
 /* local constant and vars */
 
-/**
- \ingroup ErrorPageInternal
- *
- \note  hard coded error messages are not appended with %S
- *      automagically to give you more control on the format
- */
-static const struct {
-    int type;           /* and page_id */
-    const char *text;
-}
+/// an error page (or a part of an error page) with hard-coded template text
+class HardCodedError {
+public:
+    err_type type; ///< identifies the error (or a special error template part)
+    const char *text; ///< a string literal containing the error template
+};
 
 /// error messages that cannot be configured/customized externally
-error_hard_text[] = {
-
-    {
-        ERR_SQUID_SIGNATURE,
-        "\n<br>\n"
-        "<hr>\n"
-        "<div id=\"footer\">\n"
-        "Generated %T by %h (%s)\n"
-        "</div>\n"
-        "</body></html>\n"
-    },
-    {
-        TCP_RESET,
-        "reset"
-    },
+static const std::array<HardCodedError, 7> HardCodedErrors = {
```

Comment:
We can convert this into an std::map or similar (and get rid of HardCodedError completely), but since there are just seven items, the current linear search is probably fast enough, especially since these errors are loaded/checked only once per (re)configuration.